### PR TITLE
Restore code actions order

### PR DESCRIPTION
**Pull request by pepeiborra**
An unfortunate side effect of recent changes is that the extend import suggestions ended up at the bottom of the list.

This change puts back all code actions provider together so they can be reordered more easily, and restores the previous order where extend imports would be near the top of the list

### CODE REVIEW

File: ghcide/src/Development/IDE/Plugin/CodeAction/ExactPrint.hs [hunk 1]
```diff
@@ -50,22 +49,15 @@ data Rewrite where
...
 rewriteToEdit ::
   DynFlags ->
-  Uri ->
   Anns ->
   Rewrite ->
-  Either String WorkspaceEdit
-rewriteToEdit dflags uri anns (Rewrite dst f) = do
+  Either String [TextEdit]
+rewriteToEdit dflags anns (Rewrite dst f) = do
   (ast, (anns, _), _) <- runTransformT anns $ f dflags
```
Comment (konn) [inline ghcide/src/Development/IDE/Plugin/CodeAction/ExactPrint.hs:56]: So, your suggestion in #1274  is to apply `setEntryDPT _ dp00` to the result of `f dflags` before passing ast value to `runTransformT`, right?

File: ghcide/src/Development/IDE/Plugin/CodeAction/ExactPrint.hs [hunk 2]
```diff
@@ -50,22 +49,15 @@ data Rewrite where
...
-  let editMap =
-        HMap.fromList
-          [ ( uri,
-              List
-                [ TextEdit (fromJust $ srcSpanToRange dst) $
+  let editMap = [ TextEdit (fromJust $ srcSpanToRange dst) $
                     T.pack $ tail $ exactPrint ast anns
```
Comment (konn) [inline ghcide/src/Development/IDE/Plugin/CodeAction/ExactPrint.hs:58]: If the above is the case, we also need to delete `tail` here, as it was intended to delete preceding spaces IIUC, right? (at least in #1264, this with `setEntryDPT` converts `import` into `mport`!)

File: ghcide/src/Development/IDE/Plugin/CodeAction/ExactPrint.hs (hunk 1 above)
Comment (pepeiborra) [reply ghcide/src/Development/IDE/Plugin/CodeAction/ExactPrint.hs:56]: Yes, that's right

File: ghcide/src/Development/IDE/Plugin/CodeAction/ExactPrint.hs (hunk 2 above)
Comment (pepeiborra) [reply ghcide/src/Development/IDE/Plugin/CodeAction/ExactPrint.hs:58]: Yes, that's correct

Comment (konn) [reply ghcide/src/Development/IDE/Plugin/CodeAction/ExactPrint.hs:58]: Will you make such changes also in this PR, or do you think it deserves another PR?

Comment (pepeiborra) [reply ghcide/src/Development/IDE/Plugin/CodeAction/ExactPrint.hs:58]: I think it's better left for another PR

**Review by konn — APPROVED** [review]
LGTM 👍

**Review by jneira — COMMENTED** [review]
maybe add some tests about the ordering would be great, to not break it (relative between different kinds of?)

**Comment by pepeiborra** [comment]
> maybe add some tests about the ordering would be great, to not break it (relative between different kinds of?)

Such tests would require adjustment every time a new code action is added, an existing code action is changed, etc. So I'm not keen on adding them, some tests are more trouble than they're worth.

But if anyone else wants to do it, I won't oppose either.

**Comment by jneira** [comment]
we could check at least the preferred ones are below the cut limit, but definitely not in this pr

**Pull request merged**